FeedService/Services/PiletService.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using ICSharpCode.SharpZipLib.GZip;
+using ICSharpCode.SharpZipLib.Tar;
+using System.IO;
+using System.IO.Compression;
+using System.Text.Json;
+using FeedService.Infrastructure;
+using FeedService.Models;
+using System.Threading.Tasks;
+
+namespace FeedService.Services
+{
+    public class PiletService : IPiletService
+    {
+        private readonly IStorageAccountRepository _storageAccountRepository;
+        public PiletService(IStorageAccountRepository storageAccountRepository)
+        {
+            _storageAccountRepository = storageAccountRepository;
+
+        }
+
+       /// <summary>
+       /// Gets the latest pilets from storage account.
+       /// </summary>
+       /// <returns></returns>
+        public async Task<PiletApiResponse> GetLatestPilets()
+        {
+            var latestPiletsInfo = await _storageAccountRepository.GetLatestPiletsInfoAsync();
+            var piletResponse = new PiletApiResponse();
+
+            foreach (var pilet in latestPiletsInfo)
+            {
+                var version = ExtractVersion(pilet);
+                var name = ExtractName(pilet);
+                /// This service runs locally. If you want to publish it
+                //you need to handle change this URL accordingly.
+                var url = "http://localhost:7071/api/files/" + name + "/" + version + "/index.js";
```

Comment:
For the base path we could just use an environment variable.

Usually, this is already set up properly using `Environment.GetEnvironmentVariable("WEBSITE_HOSTNAME")`.